### PR TITLE
Static analysis improvements

### PR DESCRIPTION
**Pull request by emalamela**
## Summary

Integrate static analysis tools.

We changed the philosophy of how the static analysis tools were used in this project. Instead of being something the user had to trigger manually, the current approach will force these tools to be run. The idea is for them to be automatically called in distinct moments of the development lifecycle, though not all each time. What @juanignaciomolina, @DemianIac and I discussed offline was to:

* Run styling tools, `Checkstyle` and `ktlintFormat`, in `build` or `git commit`. Since they are lightweight, they shouldn't bother the developer.
* Run bug analysis tools, `findbugs` and `ktlint`, in `git push`. Styling tools should also be called here in case the user didn't build before doing this. We discussed putting these here since they are more time consuming and would run in a more sporadic action.

Right now, **this mechanism is not implemented and I wanted to discuss some implementation aspects**. Specifically, I want to address the issue of adding the Git Hooks. Hooking calls to Git methods is done by executing commands in a `pre-$EVENT$` file that has to be placed inside `.git` folder. Said folder is ignored by git, thus losing the automatics. I think this process should be done automatically when the user clones the project, setups the project, or some similar event. I'm open to suggestions here. I though about a kickoff script or something similar, but I'd love input.

Besides all that, I fixed a couple of minor issues and update Wolmo to `2.0.4`.

## Update

The file moving mechanism was solved by the _sanity_ checks scripts that were added in the `scripts` folder. The `sanityChecks` Gradle Task, present in the file `sanity_task.gradle`, now runs in `preBuild`. That Task executes the Python script `sanity.py`: it copies the files `pre-commit` and `pre-push`, contained in `scripts/git-hooks`, to the destination `.git/hooks`.

This way, we can modify and version the hooks scripts without having to manually re-distribute the scripts.

### CODE REVIEW

**Comment by DemianIac** [comment]
@emalamela I think that you can add a folder with all the hooks and then set up git when cloning the repo (Automatically). You can set up the gitconfig file to point to that folder. We can try with one simple file if you want.

File: scripts/sanity/sanity.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        print("Git hooks step -- Copying pre-commit")
+        shutil.copy2("./git-hooks/pre-commit", "../../.git/hooks/pre-commit")
+
+    if os.path.exists("./git-hooks/pre-push"):
+        print("Git hooks step -- Copying pre-push")
+        shutil.copy2("./git-hooks/pre-push", "../../.git/hooks/pre-push")
+
+########################### Main ###############################
+def main():
+    print("Running sanity checks")
```
Comment (juanignaciomolina) [inline scripts/sanity/sanity.py]: Considering that this script is only copying the hooks and not running them, shouldn't this message display something like "Hooking sanity checks to GIT..."?

File: app/quality/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+                      value="protected" />
+        </module>
+        <module name="JavadocMethod">
+            <property name="minLineCount" value="10" />
+            <property name="scope"
+                      value="protected" />
+            <property name="allowMissingPropertyJavadoc"
+                      value="true" />
+        </module>
+        <!-- ¿JavaDoc para las variables de clases que se expongan? -->
```
[the remark this answers is not in the document]
Comment (emalamela) [reply app/quality/checkstyle/checkstyle.xml]: I agree with what you said, removing the comment.

File: app/quality/checkstyle/checkstyle.xml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        <module name="LineLength">
+            <property name="max"
+                      value="140" />
+        </module>
+        <module name="MethodLength" />
+        <module name="ParameterNumber" />
+
+        <!-- Checks for whitespace                               -->
+        <!-- See http://checkstyle.sf.net/config_whitespace.html -->
+        <!-- ¿Largo de archivo? -->
```
[the remark this answers is not in the document]
Comment (emalamela) [reply app/quality/checkstyle/checkstyle.xml]: Yeah, I don't remember why I added this question specifically but what you mentioned makes sense.

File: app/quality/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+        <module name="MultipleStringLiterals" />
+        <module name="OneStatementPerLine" />
+        <module name="NoClone" />
+        <module name="NoFinalizer" />
+        <module name="PackageDeclaration" />
+        <module name="OverloadMethodsDeclarationOrder"/>
+        <module name="SimplifyBooleanExpression" />
+        <module name="SimplifyBooleanReturn" />
+        <module name="StringLiteralEquality" />
+        <!-- ¿Depth para if, for, try, etc.? -->
```
[the remark this answers is not in the document]
Comment (emalamela) [reply app/quality/checkstyle/checkstyle.xml]: Yes. For example, something like:

```
if (someCondition) {
    for (int i = 0; i < 10; i++) {
        System.out.println("Iteration number " + i);
    }
}
```

Has a depth of 2. I was wondering what should be the limit, if any.

File: app/quality/checkstyle/checkstyle.xml [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+    <module name="SuppressionFilter">
+        <property name="file" value="${configDir}/suppressions.xml" />
+        <property name="optional" value="false"/>
+    </module>
+
+    <!-- Enables us to suppress warnings for particular lines of code via annotation. -->
+    <module name="SuppressWarningsFilter" />
+
+    <!-- ¿Limitar la cantidad de líneas por archivo? -->
```
[the remark this answers is not in the document]
Comment (emalamela) [reply app/quality/checkstyle/checkstyle.xml]: Let's add a limit of 800 for now and see how that develops.

File: scripts/sanity/sanity.py (hunk 1 above)
Comment (emalamela) [reply scripts/sanity/sanity.py]: Done!

File: app/quality/checkstyle/checkstyle.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (juanignaciomolina) [reply app/quality/checkstyle/checkstyle.xml]: I see. Maybe a limit of 3 or 4?. I guess that 4 covers pretty much every case out there.

[the remark this answers is not in the document]
Comment (emalamela) [reply app/quality/checkstyle/checkstyle.xml]: Agreed!

**Pull request merged**